clients/src/main/java/org/apache/kafka/clients/consumer/ConsumerConfig.java

```diff
@@ -222,7 +222,14 @@
     private static final String EXCLUDE_INTERNAL_TOPICS_DOC = "Whether records from internal topics (such as offsets) should be exposed to the consumer. "
                                                             + "If set to <code>true</code> the only way to receive records from an internal topic is subscribing to it.";
     public static final boolean DEFAULT_EXCLUDE_INTERNAL_TOPICS = true;
-    
+
+    /**
+     * <code>leave.group.on.close</code>
+     * Whether or not the consumer should leave the group on close. If set to <code>false</code> then a rebalance
+     * won't occur until <code>session.timeout.ms</code> expires.
+     */
+    static final String LEAVE_GROUP_ON_CLOSE_CONFIG = "leave.group.on.close";
```

Comment:
I see. Some things to think about:

1. Should we namespace internal configs in some way? Maybe an `internal` prefix?
2. It may be worth mentioning in the comment above the variable that this is an internal config and what it means (i.e. it can be changed in the future in an incompatible manner)
3. We still print the config when the configs are logged. I think this is OK, but having the `internal` prefix is important so that users don't feel compelled to use it.
4. We don't have a way to verify if the caller is internal or external, so people will still be able to use it if they really want to. Probably OK though, since the same can be said for our internal classes.